packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -246,15 +246,11 @@ class CupertinoDatePicker extends StatefulWidget {
     this.use24hFormat = false,
     this.backgroundColor,
   }) : initialDateTime = initialDateTime ?? DateTime.now(),
-       assert(mode != null),
-       assert(onDateTimeChanged != null),
-       assert(minimumYear != null),
        assert(
          minuteInterval > 0 && 60 % minuteInterval == 0,
          'minute interval is not a positive integer factor of 60',
        ),
        super(key: key) {
-    assert(this.initialDateTime != null);
```

Comment:
We need to keep these asserts for now since people may be running in a mode where null-safety is not enabled.